src/cli/cli.cpp

```diff
@@ -287,29 +287,42 @@ template <> otError Interpreter::Process<Cmd("reset")>(Arg aArgs[])
 
 void Interpreter::ProcessLine(char *aBuf)
 {
-    Arg     args[kMaxArgs + 1];
-    otError error = OT_ERROR_NONE;
+    static const char *const kCmdFactoryReset = "factoryreset";
+    Arg                      args[kMaxArgs + 1];
+    otError                  error = OT_ERROR_NONE;
 
     OT_ASSERT(aBuf != nullptr);
 
+    args[0].Clear();
+
     if (!mInternalDebugCommand)
     {
         // Ignore the command if another command is pending.
-        VerifyOrExit(!mCommandIsPending, args[0].Clear());
-        mCommandIsPending = true;
-
-        VerifyOrExit(StringLength(aBuf, kMaxLineLength) <= kMaxLineLength - 1, error = OT_ERROR_PARSE);
+        if (!mCommandIsPending)
+        {
+            mCommandIsPending = true;
+            VerifyOrExit(StringLength(aBuf, kMaxLineLength) <= kMaxLineLength - 1, error = OT_ERROR_PARSE);
+        }
+        else
+        {
+            VerifyOrExit(StringLength(aBuf, kMaxLineLength) <= kMaxLineLength - 1, args[0].Clear());
+            VerifyOrExit(ot::Utils::CmdLineParser::ParseCmd(aBuf, args, kMaxArgs) == OT_ERROR_NONE, args[0].Clear());
+            VerifyOrExit(((!args[0].IsEmpty()) && (args[0] == kCmdFactoryReset)), args[0].Clear());
+        }
     }
 
-    SuccessOrExit(error = ot::Utils::CmdLineParser::ParseCmd(aBuf, args, kMaxArgs));
-    VerifyOrExit(!args[0].IsEmpty(), mCommandIsPending = false);
+    if (args[0].IsEmpty())
+    {
+        SuccessOrExit(error = ot::Utils::CmdLineParser::ParseCmd(aBuf, args, kMaxArgs));
+        VerifyOrExit(!args[0].IsEmpty(), mCommandIsPending = false);
+    }
```

Comment:
This all seems correct but is somewhat hard to follow. 😃 

The existing code (before this PR) is also not easy to read and follow. I guess it is due to the extra `if (!mInternalDebugCommand)` checks and how we use different checks at the `exit` label:

```c++
exit:
    if ((error != OT_ERROR_NONE) || !args[0].IsEmpty())
    {
        OutputResult(error);
    }
    else if (!mCommandIsPending)
    {
        OutputPrompt();
    }
```

I don't have any immediate suggestions for the new code. It may make sense to see if the existing code can be simplified a bit (before adding the new logic to allow the "factoryreset" command).

Reply:
Just submitted this PR. Please take a look 😃 
- https://github.com/openthread/openthread/pull/11140

I think this will make it easier to allow `"factoryreset"` while `mCommandIsPending` (since we now parse the args at the top of the method, and the whole `mInternalDebugCommand` is handled separately).

Reply:
Great! I will update this commit after your commit is merged.

Reply:
I have updated this commit based on your change.